bugs/lp-openstack-ipv6-slaac-dhcpv6.txt

```diff
@@ -0,0 +1,2030 @@
++ egrep cloud-init
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@test-bionic-ipv6-dhcpv6-stateful sudo bash ./setup_proposed.sh
+  cloud-init
+Get:1 http://archive.ubuntu.com/ubuntu bionic-proposed/main amd64 cloud-init all 19.3-41-gc4735dd3-0ubuntu1~18.04.1 [411 kB]
+Preparing to unpack .../cloud-init_19.3-41-gc4735dd3-0ubuntu1~18.04.1_all.deb ...
+Unpacking cloud-init (19.3-41-gc4735dd3-0ubuntu1~18.04.1) over (19.2-36-g059d049c-0ubuntu2~18.04.1) ...
+Setting up cloud-init (19.3-41-gc4735dd3-0ubuntu1~18.04.1) ...
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@test-bionic-ipv6-dhcpv6-stateful -- dpkg-query --show cloud-init
+cloud-init	19.3-41-gc4735dd3-0ubuntu1~18.04.1
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@test-bionic-ipv6-dhcpv6-stateful -- sudo cp /etc/netplan/50-cloud-init.yaml /home/ubuntu/
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@test-bionic-ipv6-dhcpv6-stateful -- sudo cloud-init clean --logs --reboot
+Connection to test-bionic-ipv6-dhcpv6-stateful closed by remote host.
++ sleep 10
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@test-bionic-ipv6-dhcpv6-stateful -- cloud-init status --wait --long
+..................................................................................
+status: done
+time: Thu, 12 Dec 2019 18:07:24 +0000
+detail:
+DataSourceOpenStackLocal [net,ver=2]
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@test-bionic-ipv6-dhcpv6-stateful -- 'grep Trace /var/log/cloud-init*'
++ echo 'After upgrade Networking config (current -> proposed)'
+After upgrade Networking config (current -> proposed)
+++ basename /etc/netplan/50-cloud-init.yaml
++ ssh -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o LogLevel=ERROR ubuntu@test-bionic-ipv6-dhcpv6-stateful -- diff -u /home/ubuntu/50-cloud-init.yaml /etc/netplan/50-cloud-init.yaml
+--- /home/ubuntu/50-cloud-init.yaml	2019-12-12 18:06:25.040692230 +0000
++++ /etc/netplan/50-cloud-init.yaml	2019-12-12 18:06:57.204000000 +0000
+@@ -1,6 +1,6 @@
+-# This file is generated from information provided by
+-# the datasource.  Changes to it will not persist across an instance.
+-# To disable cloud-init's network configuration capabilities, write a file
++# This file is generated from information provided by the datasource.  Changes
++# to it will not persist across an instance reboot.  To disable cloud-init's
++# network configuration capabilities, write a file
+ # /etc/cloud/cloud.cfg.d/99-disable-network-config.cfg with the following:
+ # network: {config: disabled}
+ network:
+@@ -13,6 +13,7 @@
+             mtu: 8958
+             set-name: ens3
+         ens4:
++            accept-ra: true
+             dhcp6: true
+
```

Comment:
Just a question for clarification. in stateless, we already had dhcp6: true before the -proposed upgrade. Did we expect this correct behavior here pre -proposed?

Reply:
Yes, in netplan/networkd, slaac, dhcpv6-stateless, and dhcpv6-statefull all use dhcpv6;  the only key change is accept-ra: true in the stateful,  per the commit:

> commit 62bbc262c3c7f633eac1d09ec78c055eef05166a
   Date:   Wed Nov 20 18:55:27 2019 +0100

    net: IPv6, accept_ra, slaac, stateless (#51)
    
    Router advertisements are required for the default route
    to be set up, thus accept_ra should be enabled for
    dhcpv6-stateful.
    
    sysconf: IPV6_FORCE_ACCEPT_RA controls accept_ra sysctl.
    eni: mode static and mode dhcp 'accept_ra' controls sysctl.
    
    Add 'accept-ra: true|false' parameter to config v1 and
    v2. When True: accept_ra is set to '1'. When False:
    accept_ra is set to '0'. When not defined in config the
    value is left to the operating system default.
    
    This change also extend the IPv6 support to distinguish
    between slaac and dhcpv6-stateless. SLAAC is autoconfig
    without any options from DHCP, while stateless auto-configures
    the address and the uses DHCP for other options.
    
    LP: #1806014
    LP: #1808647